test/spec/sizeMapping_spec.js

```diff
@@ -206,4 +206,52 @@ describe('sizeMapping', () => {
       });
     });
   });
+
+  describe('when handling sizes defined on a bid', () => {
+    it('should filter an intersection of adUnit sizes and bid sizes', () => {
+      const testAdUnitSizes = deepClone(testSizes);
+
+      const bidOneSize = {sizes: [[728, 90]]};
+      const bidMultipleSizes = {sizes: [[728, 90], [300, 250], [300, 100]]};
+
+      // Test single valid bid size, should return single valid size
+      expect(resolveBidOverrideSizes(bidOneSize, testAdUnitSizes)).to.deep.equal(bidOneSize.sizes);
+      // Test multiple valid bid sizes, should return valid sizes
+      expect(resolveBidOverrideSizes(bidMultipleSizes, testAdUnitSizes)).to.deep.equal(bidMultipleSizes.sizes);
+    });
+
+    it('should filter bid sizes with adUnit sizes defined as a list of objects', () => {
+      // adUnit sizes defined using w/h object structure Array.<{ w:number, h:number }>
+      const testAdUnitSizesObjs = deepClone(testSizes).map(size => ({w: size[0], h: size[1]}));
+
+      const bidOneSize = {sizes: [[728, 90]]};
+      const bidMultipleSizes = {sizes: [[728, 90], [300, 250], [300, 100]]};
+
+      // Test single valid bid size, should return single valid size but defined as Array.<{ w:number, h:number }>
+      expect(resolveBidOverrideSizes(bidOneSize, testAdUnitSizesObjs, true)).to.deep.equal(bidOneSize.sizes.map(size => ({w: size[0], h: size[1]})));
+      // Test multiple valid bid sizes, should return valid sizes but defined as Array.<{ w:number, h:number }>
+      expect(resolveBidOverrideSizes(bidMultipleSizes, testAdUnitSizesObjs, true)).to.deep.equal(bidMultipleSizes.sizes.map(size => ({w: size[0], h: size[1]})));
```

Comment:
This could be a problem.  I don't think `resolveStatus` was coded to accept sizes in this format... If they're coming in like this, it probably needs to be refactored as well.